src/EnergyPlus/ReportSizingManager.cc

```diff
@@ -3577,6 +3546,90 @@ namespace ReportSizingManager {
         }
     }
 
+    Real64 setOAFracForZoneEqSizing(Real64 const &desMassFlow)
+    {
+        Real64 outAirFrac = 0.0;
+        if (desMassFlow <= 0.0) return outAirFrac;
+
+        if (DataSizing::ZoneEqSizing(DataSizing::CurZoneEqNum).ATMixerVolFlow > 0.0) {
```

Comment:
At some point, we need to break this habit of using indexes assigned external to functions.  We could easily pass in `DataSizing::CurZoneEqNum` to this function, making this function significantly less dependent on the global state.  Or even better, a reference to the entire `DataSizing::ZoneEqSizing(DataSizing::CurZoneEqNum)` object could've been passed in, nearly eliminating the dependence on global state completely (and making it much easier to test).  Imagine:

```
Real64 setOAFracForZoneEqSizing(Real64 const &desMassFlow, ZoneEqSizingData const & sizingData)
 {
     Real64 outAirFrac = 0.0;
     if (desMassFlow <= 0.0) return outAirFrac;
     if (sizingData.ATMixerVolFlow > 0.0) {
         // set central DOAS AT mixer OA fraction
         outAirFrac = min(DataEnvironment::StdRhoAir * sizingData.ATMixerVolFlow / desMassFlow, 1.0);
     } else if (sizingData.OAVolFlow > 0.0) { // set zone equipment OA fraction
         outAirFrac = min(DataEnvironment::StdRhoAir * sizingData.OAVolFlow / desMassFlow, 1.0);
     }
     return outAirFrac;
}
```

Then for testing, you would just create an instance of a ```ZoneEqSizingData``` yourself in the test, assign a value for ```DataEnvironment::StdRhoAir``` (though you could make that an argument too), and then call the function however many times you needed to test it thoroughly.

I'm not necessarily saying this needs to be done for this PR, but it's definitely a direction we want to go.  I don't see any red flags on the changes in this PR, so there probably won't need to be another commit, but if there is work to be done, this should be considered.

Reply:
I do like that and am just getting up to speed with these possibilities with the refactoring of UnitarySystem. Is there an impact on speed with this method? Or is it just cleaner? If speed, then I would like to address.